src/app/account/account-reports/account-report-setup/analysis-report-setup/analysis-report-setup.component.ts

```diff
@@ -0,0 +1,101 @@
+import { DbChangesService } from 'src/app/indexedDB/db-changes.service';
+import { AccountdbService } from 'src/app/indexedDB/account-db.service';
+import { AccountAnalysisDbService } from 'src/app/indexedDB/account-analysis-db.service';
+import { Router } from '@angular/router';
+import { IdbAnalysisItem } from 'src/app/models/idbModels/analysisItem';
+import { AnalysisDbService } from 'src/app/indexedDB/analysis-db.service';
+import * as _ from 'lodash';
+
+@Component({
+  selector: 'app-analysis-report-setup',
+  standalone: false,
+  templateUrl: './analysis-report-setup.component.html',
+  styleUrl: './analysis-report-setup.component.css'
+})
+export class AnalysisReportSetupComponent {
+  analysisReportForm: FormGroup;
+  account: IdbAccount;
+  accountAnalysisItems: Array<IdbAccountAnalysisItem>;
+  selectedReportSub: Subscription;
+  isFormChange: boolean = false;
+  selectedAnalysisItem: IdbAccountAnalysisItem;
+  itemToEdit: IdbAccountAnalysisItem;
+  facilityAnalysisItems: Array<IdbAnalysisItem> = [];
+  facilityDetails: Array<IdbAnalysisItem> = [];
+  constructor(private accountReportDbService: AccountReportDbService,
+    private accountReportsService: AccountReportsService,
+    private dbChangesService: DbChangesService,
+    private accountDbService: AccountdbService,
+    private accountAnalysisDbService: AccountAnalysisDbService,
+    private router: Router) {
+  }
+
+
+  ngOnInit() {
+    this.account = this.accountDbService.selectedAccount.getValue();
+    this.selectedReportSub = this.accountReportDbService.selectedReport.subscribe(val => {
+      if (!this.isFormChange) {
+        this.analysisReportForm = this.accountReportsService.getAnalysisFormFromReport(val.analysisReportSetup);
+        this.setAnalysisOptions(val);
+      } else {
+        this.isFormChange = false;
+      }
+    });
+  }
+
+  ngOnDestroy() {
+    this.selectedReportSub.unsubscribe();
+  }
+
+  async save() {
+    this.isFormChange = true;
+    this.setSelectedAnalysisItem();
+
+    let selectedReport: IdbAccountReport = this.accountReportDbService.selectedReport.getValue();
+    selectedReport.analysisReportSetup = this.accountReportsService.updateAnalysisReportFromForm(selectedReport.analysisReportSetup, this.analysisReportForm);
+    if (this.selectedAnalysisItem) {
+      selectedReport.baselineYear = this.selectedAnalysisItem.baselineYear;
+    }
+    await firstValueFrom(this.accountReportDbService.updateWithObservable(selectedReport));
+    await this.dbChangesService.setAccountReports(this.account);
+    this.accountReportDbService.selectedReport.next(selectedReport);
+  }
+
+  setAnalysisOptions(report: IdbAccountReport) {
+    let analysisOptions: Array<IdbAccountAnalysisItem> = this.accountAnalysisDbService.accountAnalysisItems.getValue();
+    this.accountAnalysisItems = analysisOptions.filter(option => { return option.reportYear == report.reportYear && option.energyIsSource });
```

Comment:
We don't need to filter out the "energyIsSource" options here. We are actually needing to take out that filter in other places.